application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -182,4 +178,37 @@ public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
         throw new UnsupportedOperationException("Not used");
     }
 
+    private boolean isPostAuthor(Member interactionUser, Message message) {
+        if (message.getEmbeds().isEmpty())
+            return false;
+
+        String embedAuthor = Objects
+            .requireNonNull(message.getEmbeds().get(0).getAuthor(),
+                    "embed author for forum post is null")
+            .getName();
+
+        return embedAuthor.equals(interactionUser.getUser().getName());
+    }
+
+    private boolean isAuthorized(Member interactionUser, ThreadChannel channel,
+            Message forumPostMessage) {
+        return (channel.getOwnerIdLong() == interactionUser.getIdLong())
+                || helper.hasTagManageRole(interactionUser)
+                || isPostAuthor(interactionUser, forumPostMessage);
+    }
+
+    private void handleDismiss(Member interactionUser, ThreadChannel channel,
+            Message forumPostMessage, ButtonInteractionEvent event, List<String> args) {
+        boolean isAuthorized = isAuthorized(interactionUser, channel, forumPostMessage);
+        if (!isAuthorized) {
+            event.getHook().sendMessage("Not allowed").setEphemeral(true).queue();
```

Comment:
the `Not allowed` reads a bit odd. perhaps `Sorry, you do not have permission to dismiss this message.`

Reply:
Hehe, was meant to be testing only. Will do.